implementation/exporters/src/main/java/io/smallrye/opentelemetry/implementation/exporters/AbstractVertxExporterProvider.java

```diff
@@ -57,14 +67,34 @@ protected HttpExporter<T> createHttpExporter(ConfigProperties config, String htt
                 false);//TODO: this will be enhanced in the future
     }
 
+    /**
+     * If the CDI qualifier is specified in the config, it tries to get it from CDI, and if CDI does not provide such
+     * an instance on the specified qualifier, it will log some WARNING messages and return a new Vertx instance.
+     * If the CDI qualifier is not specified in the config, it creates a new Vertx instance.
+     */
+    private Vertx getVertx(ConfigProperties config) {
+        String cdiQualifier = config.getString(OTEL_EXPORTER_VERTX_CDI_QUALIFIER);
+        if (cdiQualifier != null && !cdiQualifier.isEmpty()) {
+            Instance<Vertx> vertxCDI = CDI.current().select(Vertx.class, Identifier.Literal.of(cdiQualifier));
+            if (vertxCDI != null && vertxCDI.isResolvable()) {
+                return vertxCDI.get();
+            } else {
+                logger.log(Level.WARNING, "The given CDI qualifier \"{0}\" for Vertx instance is not resolvable.",
+                        cdiQualifier);
+            }
+        }
+        logger.log(Level.INFO, "Create a new Vertx instance");
```

Comment:
It is my personal opinion that we should try to resolve `Vertx` from CDI without a qualifier here. If that doesn't resolve, I'm not sure, but creating a new instance unconditionally doesn't seem like the worst idea.

Reply:
I would prefer to just try to resolve it when a qualifier is specified, so it is on smallrye-opentelemetry's side to decide which one to use. If we try to resolve it from CDI without a qualifier, as long as there is a Vertx instance exposed in CDI container, it is forced to get injected to smallrye-opentelemtry.

Reply:
Sounds like exactly what I would expect :shrug:

Reply:
As it's written now, if there's an @Identifer defined but no bean properly annotated, we log a WARN and fall back to explicit Vertx instantation. If I understand you correctly, @Ladicek, you would like to put an attempt to find and unqualified Vertx bean in the middle before we fall back, correct? If that's so, then the logic would look like:

1. Check for qualified
2. Check for unqualified
3. Return local instance

That actually does seem like a safer and easier approach. *Requiring* the extra annotation, even in instances where it's not strictly needed, seems like extra work for no real value. The qualifier's intent is to allow the system to properly select Vertx beans in scenarios when there's more than one. If there's only one, then it's obvious which to select. If that bean is NOT what the user wants, then he or she should provide a properly annotated option.